maven-core/src/main/java/org/apache/maven/lifecycle/internal/DefaultLifecycleExecutionPlanCalculator.java

```diff
@@ -554,6 +554,8 @@ private void injectLifecycleOverlay( Map<String, List<MojoExecution>> lifecycleM
 
         MojoExecution forkedExecution = new MojoExecution( forkedMojoDescriptor, forkedGoal );
 
+        forkedExecution.setLifecyclePhase( mojoExecution.getLifecyclePhase() );
```

Comment:
how does that relate to MNG-7129 and the switch from JAXB to Modello?

Reply:
It does not relate to JAXB, however, this fixes a NPE which happens in the cache system when using forked lifecycles.  It may be fixed in a different way, but it did look like the easier fix.
My reasoning is the following : I suppose no-one ever looked for the phase of forked plugin executions and it makes sense to me to have those phases set to the phase of the original mojo execution.
The cache system does look and uses the phase in several locations, and this caused NullPointerException.

Reply:
Hi
Thanks for the change! Jaxb is gone for the good. In overall pr looks good for me.
Few notes: 
1. as i understand modello schema is located in core. if we move the cache to an extension in future - doest it mean there will be dependency on core from the extension? Looks like modello scheme actuall couples the cache and the core.
2 There should be few small jaxb related leftovers - dependency in pom itself and in licenses, but in general it looks good.

Reply:
@AlexanderAshitkin I don't see why the modello generated code could not be moved into the extension.